core/gpc_api.php

```diff
@@ -384,31 +389,77 @@ function gpc_set_cookie( $p_name, $p_value, $p_expire = false, $p_path = null, $
 		$p_domain = config_get_global( 'cookie_domain' );
 	}
 
-	return setcookie( $p_name, $p_value, $p_expire, $p_path, $p_domain, $g_cookie_secure_flag_enabled, $p_httponly );
+	if( null === $p_samesite ) {
+		$p_samesite = config_get_global( 'cookie_samesite' );
+	}
+
+	if (PHP_VERSION_ID < 70300) {
+		# Take advantage of PHP bug #69948 (fixed in PHP 7.3) to inject the
+		# Samesite attribute in the cookie's path
+		if( $p_samesite ) {
+			$p_path .= '; samesite=' . $p_samesite;
+		}
+		return setcookie( $p_name, $p_value, $p_expire, $p_path, $p_domain, $g_cookie_secure_flag_enabled, $p_httponly );
+	}
+
+	# Use new function signature available since PHP 7.3
+	$t_options = array(
+		'expires' => $p_expire,
+		'path' => $p_path,
+		'domain' => $p_domain,
+		'samesite' => $p_samesite,
+		'secure' => $g_cookie_secure_flag_enabled,
+		'httponly' => $p_httponly,
+	);
+	return setcookie( $p_name, $p_value, $t_options );
 }
 
 /**
  * Clear a cookie variable
- * @param string $p_name   Cookie clear to set.
- * @param string $p_path   Cookie path.
- * @param string $p_domain Cookie domain.
+ * @param string $p_name     Cookie clear to set.
+ * @param string $p_path     Cookie path.
+ * @param string $p_domain   Cookie domain.
+ * @param string $p_samesite SameSite attribute.
  * @return boolean
  */
-function gpc_clear_cookie( $p_name, $p_path = null, $p_domain = null ) {
+function gpc_clear_cookie( $p_name, $p_path = null, $p_domain = null, $p_samesite = null ) {
 	if( null === $p_path ) {
 		$p_path = config_get_global( 'cookie_path' );
 	}
 	if( null === $p_domain ) {
 		$p_domain = config_get_global( 'cookie_domain' );
 	}
+	if( null === $p_samesite ) {
+		$p_samesite = config_get_global( 'cookie_samesite' );
+	}
 
 	if( isset( $_COOKIE[$p_name] ) ) {
 		unset( $_COOKIE[$p_name] );
 	}
 
 	# don't try to send cookie if headers are send (guideweb)
 	if( !headers_sent() ) {
-		return setcookie( $p_name, '', -1, $p_path, $p_domain );
+		# Note: This setcookie() call triggers a console warning in Firefox:
+		# Cookie “MANTIS_collapse_settings” has been rejected because it is already expired.
```

Comment:
Still using the former hardcoded name

Reply:
This is just a comment reflecting the cookie's default name as it appears in the Firefox console... Or am I missing something ?

Reply:
I replaced `MANTIS` by `<PREFIX>` in the comment, assuming that's what you meant. If not, let me know.